src/main/fc/fc_msp.c

```diff
@@ -1136,16 +1137,32 @@ static bool mspFcProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst, mspPostProcessFn
         break;
 
     case MSP_TRANSPONDER_CONFIG:
+    {
 #ifdef TRANSPONDER
-        sbufWriteU8(dst, 1); //Transponder supported
+        uint8_t header = 1; //transponder supported
+
+        switch(transponderConfig()->provider){
+            case ILAP:
+               header |= 0x02;
+               break;
+            case ARCITIMER:
+                 header |= 0x04;
+                 break;
+            default:
+                break;
+        }
+
+        header |= (sizeof(transponderConfig()->data) << 4);
+
```

Comment:
It's not clear from the code what 1, 0x02 and 0x04 and the << 4 do.

if you have a look around you'll see we use `#defines` for this kind of thing.

if we add 'NONE' to the transponder provider list we can simply use that rather than the switch statement here.  using 'NONE' is an approach we have starting taking recently for quite a few things, such as blackbox device, voltage meter source, current meter source, etc.  it simplifies things quite a bit.

I'll merge this as-is and then fix it.